solution/task2_pca.py

```diff
@@ -0,0 +1,124 @@
+import time
+t=time.time()
+
+import numpy as np
+import pandas as pd
+from sklearn.svm import SVC
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import StandardScaler
+from sklearn.neural_network import MLPClassifier
+from sklearn.svm import SVC
+from sklearn.gaussian_process import GaussianProcessClassifier
+from sklearn.gaussian_process.kernels import RBF
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.ensemble import RandomForestClassifier, AdaBoostClassifier
+from sklearn.naive_bayes import GaussianNB
+from sklearn.feature_selection import mutual_info_classif
+import matplotlib.pyplot as plt
+from sklearn.decomposition import PCA
+
+#Assign column name
+columns = ['F' + str(i) for i in range(130)]
+
+#Read data and store in a DataFrame
+data = pd.read_csv('../data/data.csv', sep=' ', header=None, names = columns)
+target = pd.read_csv('../data/target.csv', header=None)
+
+
+#Convert DataFrame to an array
+X = data.to_numpy()
+y = target.to_numpy().flatten()
+
+# Separate data into training and validation sets
+X_train, X_valid, y_train, y_valid = train_test_split(X, y, test_size=0.2, random_state=42, stratify=y)
+
+#Feature Normalization (data standardization) of train and test data using only TRAINING DATA for both Training set and Validation Set,
+#to sure our model generalize well on new, unseen data.
+
+scaler = StandardScaler()
+scaler.fit(X_train)
+
+X_train = scaler.transform(X_train)
+X_valid = scaler.transform(X_valid)
+
+#Define function for Mutual Information (MI) Score
+def make_mi_scores(X, y):
+    mi_scores = mutual_info_classif(X, y)
+    mi_scores = pd.Series(mi_scores, name="MI Scores", index=X.columns)
+    mi_scores = mi_scores.sort_values(ascending=False)
+    return mi_scores
+
+# Create principal components both for training and validation data using Training Data
+#to sure our model generalize well on new, unseen data.
+pca = PCA()
+pca.fit(X_train)
+
+X_pca = pca.transform(X_train)
+X_valid_pca = pca.transform(X_valid)
+
+# Convert to dataframe train data
+component_names = [f'PC{i+1}' for i in range(X_pca.shape[1])]
+X_pca = pd.DataFrame(X_pca, columns=component_names)
+
+# Convert to dataframe valid data
+component_names = [f'PC{i+1}' for i in range(X_valid_pca.shape[1])]
+X_valid_pca = pd.DataFrame(X_valid_pca, columns=component_names)
+
+#Calculate MI Scores
+mi_scores_pca = make_mi_scores(X_pca, y_train, discrete_features=False)

```

Comment:
I forgot hear to delete third parameter "discrete_features=False". Please delete it before running the code.

Reply:
Can you do it yourself and submit again?

Reply:
It's done!